pkg/controller/build/schedule_pod.go

```diff
@@ -135,6 +156,9 @@ func newBuildPod(build *v1alpha1.Build, operatorImage string) *corev1.Pod {
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: build.Namespace,
 			Name:      buildPodName(build.Spec.Meta),
+			Labels: map[string]string{
+				"camel.apache.org/build": build.Name,
+			},
 		},
 		Spec: corev1.PodSpec{
 			ServiceAccountName: "camel-k-operator",
```

Comment:
I wonder if that'd make sense we create a dedicated SA for build pods. I think they don't need the full permissions, for example pod creation, that can be seen as a security risk: https://kubernetes.io/docs/reference/access-authn-authz/authorization/#privilege-escalation-via-pod-creation. It's not directly related to this PR, so more a note for future work.

Reply:
Good point. Ok, if we create a reduced service account for builds, then we better use it in same namespace and so we need the previous permissions :)
(Future usage is actual)

Reply:
I've added a reduced sa for builder.. Tested on minikube.. Now I need to test it on openshift.. too much manual testing

Reply:
Ok seems to work also on openshift :smile: